recipes/minizip/all/conanfile.py

```diff
@@ -0,0 +1,156 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class MinizipConan(ConanFile):
+    name = "minizip"
+    description = "minizip is a zip manipulation library written in C "
+    "that is supported on Windows, macOS, and Linux."
+    topics = ("conan", "minizip", "compression")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/nmoinvaz/minizip"
+    license = "Zlib"
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    generators = "cmake_find_package"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_zlib": [True, False],
+        "with_bzip2": [True, False],
+        "with_zstd": [True, False],
+        "with_openssl": [True, False],
+        "with_lzma": [True, False],
+        "with_libbsd": [True, False],
```

Comment:
Check the openssl option of cmake to see how settings-dependent options can be handled:
https://github.com/conan-io/conan-center-index/commit/2931d4d4071fcb48ab884f0861fb9f8e2318b6e3#diff-b44e21714eb78e422c3ab9e8ede8c7a3

In this case, you can:
1. set `options = { ..., with_libbsd: ["auto", True False], ... }`
2. set `default_options = { ..., with_libbsd: "auto" ... }`
3. add a `_with_libbsd` property function:
    ```python
    def _with_libbsd(self):
         if self.options.with_libbsd == "auto":
            return self.settings.os != "Windows"
        else:
            return bool(self.options.with_libbsd)
    ```
4. require libbsd optionally in `requirements`:
    ```python
    def requirements(self):
        if self._with_libbsd:
            self.requires("libbsd/0.10.0")
    ```
5. use the `self._with_libbsd` function everywhere, also no need any more for `_is_unix_like`. I think.
6. do this in `package_id` to make sure that the package id of "auto" is not possible:
    ```python
    def package_id(self):
        self.info.with_libbsd = self._with_libbsd
    ```

Reply:
Thanks for pointing out this trick